### ISSUE TITLE
[Research] How to handle file path changes in `i18n` counterparts

### ISSUE THREAD

**Issue by roadlittledawn** [newrelic/docs-website#4232]
## Summary

Right now if a PR contains changes that include moving a file from directory to another (which is a status of `removed` from git's perspective), we check to see if the file has counterparts in the `src/i18n` language dirs and if it does, we throw a warning as part of the `validate-pr` workflow check.

We haven't really run into this case yet since that check was put in place. But we will. We should figure out how we want to handle these changes when the check fails.

## Outstanding questions
- Is this warning sufficient to kick off a human process for PR submitter / tech docs teammember to make equivalent change?
- Should a writer move/remove equivalent `i18n` file at all?
- Is it okay from a translation workflow perspective to manually change filepaths in `i18n`?

## Examples
This doc was moved and our translation workflow check probably wasn't in place, so when we it was moved and subsequent PRs made, we created a new file at the new path and never removed the old one from i18n.
* https://github.com/newrelic/docs-website/blob/develop/src/i18n/content/jp/docs/accounts/accounts-billing/account-setup/account-id.mdx
* https://github.com/newrelic/docs-website/blob/develop/src/i18n/content/jp/docs/accounts/accounts-billing/account-structure/account-id.mdx

## Acceptance criteria
- [x] Determine if warning is sufficient for tech docs / PR submitter to make equivalent change in `i18n`
- [X] If not, file ticket with some technical specs on how we can do this programmatically 

## Helpful links
https://github.com/newrelic/docs-website/issues/5127

**Comment by roadlittledawn** [comment]
Long term, Julien said it may make sense to create a JSON file that records what files were moved in en / what should be removed in i18n, so he / his TBD teammate can go in and do that in subsequent PRs.

worth noting that since the human translations aren't added to site right away, we don't want to delete i18n files until the new i18n files come back from the file move in english.

**Comment by roadlittledawn** [comment]
Chatted w/ @nr-kkenney about this a bit. Some notes:

Once MT is in place for Japanese, it's much more likely content folks will move files and the workflow check will warn them that equivalent `i18n` files need to be updated.

<details>
<summary>Short term solution</summary>
Content authors apply same file moves to corresponding `i18n` files.

This ensures that when PR is merged to `develop` and relevant files are added to translation queue, the English redirects and URL paths resolve to their Japanese translated equivalents.

When translations eventually come back in a PR (which could be up to 14 days for human, 2 days machine), the PR will be to modify the files at the new path. Currently, when they come back, new files are written at the new path and the old files are not deleted.
</details>
<details>
<summary>Long term proposal</summary>
We will file a new ticket to be included in Q4 Translation Scaling epic to attempt to do this programmatically.

How much can be automated depends on how the github API distinguishes files that are deleted VS files that were moved (file path changed) in a given PR.

We'd like to try and automate the deletion and moving of corresponding `i18n` files during a workflow that runs on PR, so content authors don't have the burden of making the same file moves per language we support (up to 6 in Q4)
</details>

**Comment by roadlittledawn** [comment]
@austin-schaefer will ping @zuluecho9 re: working w/me around updating style guide on how to move/remove files to include info on apply same changes to affected `i18n` files

**Comment by zuluecho9** [comment]
@roadlittledawn If it helps, here's a synopsis of my understanding for the required procedures for the i18n/content/jp folders. If you can confirm this makes sense, I can try to put this into our style guide. (and if I'm off base maybe we meet up on it to discuss more): 

If you're manipulating the docs site folder structure (e.g., moving files between folders; moving folders; renaming folders, deleting folders) or moving or renaming mdx files, AND if the manipulated folders or files exist in the i18n/content/jp directory, you must perform the same operations to the folders and files in the i18n directory. Put another way: any folder or file manipulation in the `content/docs` directory that is NOT editing the actual content of a file must have the equivalent operations performed in the i18n directory. 

In practice, some folders and files will not exist in the i18n directory because the system only creates mdx files and the folders that contain them if an mdx file is marked to be translated (the 'translate - jp' field). If the folders or files don't exist in the i18n directory, there is no action to take for those folders or files. 

One method for implementing the same folder/file manipulation is to place two finder windows side by side and mirror each movement you take in the `docs` directory in the `i18n` directory. For example, a sample workflow for moving a folder of docs from one directory to another might look like this: 1) add the necessary redirects for the mdx files you'll be moving, 2) in your finder window, in the `docs` directory, drag the folder from one place to the desired location, 3) in the finder window, in the `i18n` directory, do that same drag and drop.
